### PR TITLE
RESOURCE-238 AWS::ApiGateway::VpcLink

### PR DESCRIPTION
**Pull request by soumyo13**
Signed-off-by: Soumyodeep Karmakar <soumyo.k13@gmail.com>

### Description

1. It contains all the singular and plural resources.
2. Added the resource_id in the singular resource.
3. Modified the docs according to the new format. 

### Issues Resolved
This is a new resource. There are no issues to be fixed.

### Check List
Please fill a box or appropriate ([x]) or mark N/A.
- [x] New functionality includes integration tests/controls
- [x] New Terraform resources
- [x] Documentation provided or updated for resources 
- [x] All Integration Tests pass
- [x] All Unit Tests pass
- [x] `rake lint` passes
- [x] All commits have been signed off for the Developer Certificate of Origin. See <https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco>

### CODE REVIEW

**Review by sa-progress — APPROVED** [review]

File: docs-chef-io/content/inspec/resources/aws_api_gateway_vpc_link.md [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+```
+
+## Parameters
+
+`vpc_link_id` _(required)_
+
+: The identifier of the VpcLink.
+
+## Properties
+
```
Comment (dkumaras) [inline docs-chef-io/content/inspec/resources/aws_api_gateway_vpc_link.md:38]: @soumyo13 - Do we need to include the field types of all these properties?

File: docs-chef-io/content/inspec/resources/aws_api_gateway_vpc_link.md [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+`status`
+: The status of the VPC link. The valid values are AVAILABLE , PENDING , DELETING , or FAILED . Deploying an API will wait if the status is PENDING and will fail if the status is DELETING .
+
+`status_message`
+: A description about the VPC link status.
+
+`tags`
+: The collection of tags. Each tag element is associated with a given resource.
+
```
Comment (dkumaras) [inline docs-chef-io/content/inspec/resources/aws_api_gateway_vpc_link.md:59]: Shall we mention this to the tags: The valid character set is [a-zA-Z+-=._:/]. The tag key can be up to 128 characters and must not start with aws:. The tag value can be up to 256 characters.

**Review by dkumaras — COMMENTED** [review]
@soumyo13 - Pls review my suggestions and let me know your thoughts.

File: docs-chef-io/content/inspec/resources/aws_api_gateway_vpc_link.md (hunk 1 above)
Comment (soumyo13) [reply docs-chef-io/content/inspec/resources/aws_api_gateway_vpc_link.md:38]: No.

File: docs-chef-io/content/inspec/resources/aws_api_gateway_vpc_link.md (hunk 2 above)
Comment (dkumaras) [reply docs-chef-io/content/inspec/resources/aws_api_gateway_vpc_link.md:59]: ok

**Review by dkumaras — APPROVED** [review]

**Pull request merged**